compiler/optimizer/LoopCanonicalizer.cpp

```diff
@@ -1999,6 +2073,39 @@ void TR_LoopCanonicalizer::eliminateRedundantInductionVariablesFromLoop(TR_Regio
                      }
                   }
                }
+            static bool disableTrivialArrayOperationCheck = (NULL != feGetEnv("TR_DisableTrivialArrayOperationCheck"));
+            bool istrivialArrayOperation = false;
+            /* Since we know that trivalArrayOperationCheck will return false for
+             * improper regions we can skip the check all together if the region
+             * is a acyclic
+             */
+            if (!disableTrivialArrayOperationCheck && !naturalLoop->isAcyclic())
+               {
+               traceMsg(comp(),"Starting the trivial array operation check");
+               TR::Node *derivedInductionVarNode = NULL;
+               TR_RegionStructure::SubNodeList::iterator nodeIT = naturalLoop->begin();
+               TR_Structure *pred = toStructureSubGraphNode((*(nodeIT)))->getStructure();
+               for (TR::Block *derivedInductionVarBlock = pred->getEntryBlock();
+                    !derivedInductionVarNode && !derivedInductionVarBlock->isEndBlock();
+                    derivedInductionVarBlock = derivedInductionVarBlock->getNextBlock())
+                  {
+                  for (TR::TreeTopIterator treeTopIt(derivedInductionVarBlock->getEntry(),comp());
+                       !derivedInductionVarNode && treeTopIt != derivedInductionVarBlock->getExit();
+                       treeTopIt.stepForward())
+                     {
+                     if (treeTopIt.currentNode()->getSymbol() == derivedInductionVar->getSymbol())
+                        derivedInductionVarNode = treeTopIt.currentNode();
+                     }
+                  }
+
+               TR_ASSERT_FATAL_WITH_NODE(derivedInductionVarNode, derivedInductionVarNode != NULL, "Unable to find node containing the derived induction variable");
+               TR::NodeChecklist visited(comp());
+               istrivialArrayOperation = trivialArrayOperationCheck(comp(),visited, derivedInductionVarNode);
+               if (istrivialArrayOperation)
+                  performTransformation(comp(), "%sSkipping induction variable elimination for %d as it is being used in a trivial array operation.\n",OPT_DETAILS, derivedInductionVar->getReferenceNumber());
```

Comment:
Check other uses of `performTransformation`. This API is used to guard transformations which can then be searched via `lastOptTransformationIndex` or `lastOptSubindex`. In your current implementation we will proceed with the transformation irregardless because no code is being guarded by the return value of `performTransformation`. We need to change this to something like:

```
if (istrivialArrayOperation)
   istrivialArrayOperation = performTransformation(comp(), "%sSkipping...
```

Reply:
ah I see your point. But if we disable `trivialArrayOperationCheck` opt using `lastOptTransformationIndex` or `lastOptSubindex`. That would also disable the redundant var elimination transformation here https://github.com/eclipse/omr/blob/53ce7df9200f1e308c8ad2d5a12dd8795b61fd14/compiler/optimizer/LoopCanonicalizer.cpp#L2148  since they are mutually exclusive.

I think we should look at the `trivialArrayOperationCheck` not as a transformation but rather as a input/criteria used to evaluate cost/benefit of redundant var elimination.

Reply:
Skipping the transform needs to be exposed to the triage/binary search process somehow so you need to consider that and do something appropriate here, whatever that is.

Reply:
Thinking about it now, it's definitely better to disable both transformations rather than disable one only to find out other was enabled the entire time.